### ISSUE TITLE
Hikari CP connections disappear from pool

### ISSUE THREAD

**Issue by arumugham86** [brettwooldridge/HikariCP#1332]
#### Environment
```
HikariCP version: 2.5.1
JDK version     : 1.8.0_111
Database        : Maria DB
Driver version  : 2.1.0
```

Idletimeout is 30s, Max Life time is 1800s, minIdle is 5, maxConn is 50, connTimeout is 5s

----

I am not able to explain the behaviour exhibited via following logs, where did all the 40+ connections go in 2s? Why?

2019-02-27 14:25:56.951 DEBUG 25 -- [iCP housekeeper] com.zaxxer.hikari.pool.HikariPool        : XXX - Pool stats (total=50, active=49, idle=1, waiting=0)
2019-02-27 14:25:58.926 DEBUG 25  -- [iCP housekeeper] com.zaxxer.hikari.pool.HikariPool        : XXX - Pool stats (total=5, active=0, idle=5, waiting=0)

Is there a bug in this version? a possible leak (we are investigating)? parallel multithreaded calls vs any known problems in Hikari? are we missing any further configuration?

**Comment by brettwooldridge** [comment]
@arumugham86 Multiple threads should probably not share a Connection at the same time, especially if it is unclear which one owns the responsibility for closing it.

Having said that, if you are on Java 8 there is no reason not to be on the latest version of HikariCP.  Please update as literally dozens and dozens of issues have been fixed since v2.5.1 was released.

**Comment by arumugham86** [comment]
@brettwooldridge thanks, we are doing that. And, I found the answer to my question, we were using two CP with same pool name and two lines are for one pool each! {facepalm}
